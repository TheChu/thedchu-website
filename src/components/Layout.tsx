import { graphql, useStaticQuery } from 'gatsby';
import React, { FC, ReactElement, ReactNode } from 'react';
import Helmet from 'react-helmet';
import '../assets/sass/creative.scss';

interface LayoutProps {
  children: ReactNode;
}

const Layout: FC<LayoutProps> = (props: LayoutProps): ReactElement => {
  const { children } = props;
  const {
    site: {
      siteMetadata: { title },
    },
  } = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `);

  return (
    <>
      <Helmet
        title={title}
        meta={[
          { name: 'description', content: 'thedchu personal website' },
          { name: 'keywords', content: 'site, web' },
        ]}
      >
        <html lang="en" />
      </Helmet>
      <>{children}</>
    </>
  );
};

export default Layout;
