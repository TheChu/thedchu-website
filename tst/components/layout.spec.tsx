import { render } from '@testing-library/react';
import { useStaticQuery } from 'gatsby';
import React from 'react';
import Helmet from 'react-helmet';
import Layout from '../../src/components/Layout';

(useStaticQuery as jest.Mock).mockReturnValue({
  site: { siteMetadata: { title: 'thedchu' } },
});

describe('Layout', () => {
  it('should match Layout snapshot', () => {
    const { asFragment } = render(
      <Layout>
        <div className="test div" />
      </Layout>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render metadata', () => {
    render(
      <Layout>
        <div className="test div" />
      </Layout>
    );
    const { htmlAttributes, metaTags, title } = Helmet.peek();
    expect(title).toBe('thedchu');
    expect(metaTags.length).toBe(2);
    expect(metaTags).toEqual(
      expect.arrayContaining([
        { name: 'description', content: 'Creative' },
        { name: 'keywords', content: 'site, web' },
      ])
    );
    expect(htmlAttributes).toEqual({ lang: 'en' });
  });
});
