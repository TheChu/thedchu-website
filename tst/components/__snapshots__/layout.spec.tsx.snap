// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match Layout snapshot 1`] = `
<DocumentFragment>
  <div
    class="test div"
  />
</DocumentFragment>
`;
